subsys/net/lib/mqtt/mqtt.c

```diff
@@ -591,11 +592,16 @@ int mqtt_live(struct mqtt_client *client)
 	if ((client->keepalive > 0) &&
 	    (elapsed_time >= (client->keepalive * 1000))) {
 		err_code = mqtt_ping(client);
+		ping_sent = true;
 	}
 
 	mqtt_mutex_unlock(client);
 
-	return err_code;
+	if (ping_sent) {
+		return err_code;
+	} else {
+		return -EAGAIN;
```

Comment:
Would be nice to mention in the `mqtt_live` API documentation when `-EAGAIN` can be returned and that it's not an actual error.